proxy/src/telemetry/metrics/histogram.rs

```diff
@@ -77,23 +80,46 @@ impl<V: Into<u64>> Histogram<V> {
             .expect("all values must fit into a bucket");
 
         self.buckets[idx].incr();
-        self.sum += Wrapping(value);
+        self.sum += value;
     }
+}
+
+impl<V: Into<u64>> FmtMetric for Histogram<V> {
+    fn fmt_metric<N: Display>(&self, f: &mut fmt::Formatter, name: N) -> fmt::Result {
+        let mut total = Counter::default();
+        for (le, count) in self.bounds.0.iter().zip(self.buckets.iter()) {
```

Comment:
Doesn't `&'a Histogram` already have an `IntoIterator` impl that does exactly the same thing?
I think this can be just

```rust
for (le, count) in self {
    // ...
}
```

Reply:
Also, this could probably be a `fold` (but I know you dislike side-effecting folds so TIOLI).